docs/skopeo.1.md

```diff
@@ -192,6 +192,48 @@ Verify a signature using local files, digest will be printed on success.
 
 **Note:** If you do use this, make sure that the image can not be changed at the source location between the times of its verification and use.
 
+## skopeo sync
+**skopeo copy** [**--source** _source_] [**--source-file** _source-file_] _destination_
+
+Copy all the images from _source_ (or _source-file_) to _destination_.
+
+Useful to keep in sync a local docker registry mirror. It can be used to populate also registries running inside of air-gapped environments.
+
+_source_ can be either a repository hosted on a docker registry (eg: docker://docker.io/busybox) or a local directory (eg: dir:///media/usb/).
+
+_source-file_ is a YAML file with a set of source images from different docker registry. Local directory are not supported.
+
+**skopeo sync** will copy all the tags of an image when _source_ uses the docker://' transport and no tag is specified.
+
+_destination_ can be either a docker registry (eg: docker://my-registry.local.lan) or a local directory (eg: dir:///media/usb).
```

Comment:
The PR introduction says

> The image namespace is changed during the `docker://` to `docker://` or `dir:` copy. The FQDN of the registry hosting the image will be added as new root namespace of the image. For example, the image `registry.example.com/busybox:latest` will be copied to `registry.local.lan/registry.example.com/busybox:latest`

This inflexibly hard-codes policy, and seems entirely unnecessary: whatever script is used for the regular sync can just specify `registry.local.lan/registry.example.com/busybox` as the destination.

Also, this policy forces `sync` to invent an entirely new `docker://hostname` syntax, which is ambiguous with the image name syntax used everywhere else, and has a _different meaning_ (the image `docker://hostname` == `docker://docker.io/library/hostname:latest`).

… and AFAICT it s not documented in this man page.

Reply:
Some kind of scoping is necessary when running `sync` from a YAML file, as you may be pulling the same image name from different namespaces or domains.

The reason to have this kind of scoping is that the original image location can be easily extrapolated, and this information is useful when running a server that is configured to mirror multiple registries.

Maybe the requirement can be dropped when source references to a `docker://` location (and a flag can be added to make the current behaviour optional)

Reply:
> Some kind of scoping is necessary when running sync from a YAML file, as you may be pulling the same image name from different namespaces or domains.

My first instinct is to specify the destination in the YAML as well, then, not to hard-code the naming in software.

It’s not obvious to me why `sync --source-yaml $src $dest` would have the sources specified in a (presumably mostly fixed) config file but the destination in a (presumably mostly changing) destination.

If a user sets up a regular (e.g. weekly) sync process, won’t both the source and the destination the same for all runs (every week the same servers and same destinations)? (My best guess is that when the destination is on a removable medium, the physical device, and thus presumably the mount path, may change each time — but aren’t mounts often named by the volume label, which is always the same?)